ntoskrnl/se/debug.c

```diff
@@ -0,0 +1,388 @@
+/*
+ * PROJECT:         ReactOS Kernel
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Security subsystem debug routines support
+ * COPYRIGHT:       Copyright 2022 George Bișoc <george.bisoc@reactos.org>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* PRIVATE FUNCTIONS **********************************************************/
+
+/**
+ * @brief
+ * Converts an Access Control Entry (ACE) type to a string.
+ *
+ * @return
+ * Returns a converted ACE type strings. If no
+ * known ACE type is found, it will return
+ * UNKNOWN TYPE.
+ */
+static
+PCSTR
+SepGetAceTypeString(
+    _In_ UCHAR AceType)
+{
+#define TOSTR(x)    #x
+    static const PCSTR AceTypes[] =
+    {
+        TOSTR(ACCESS_ALLOWED_ACE_TYPE),
+        TOSTR(ACCESS_DENIED_ACE_TYPE),
+        TOSTR(SYSTEM_AUDIT_ACE_TYPE),
+        TOSTR(SYSTEM_ALARM_ACE_TYPE),
+        TOSTR(ACCESS_ALLOWED_COMPOUND_ACE_TYPE),
+        TOSTR(ACCESS_ALLOWED_OBJECT_ACE_TYPE),
+        TOSTR(ACCESS_DENIED_OBJECT_ACE_TYPE),
+        TOSTR(SYSTEM_AUDIT_OBJECT_ACE_TYPE),
+        TOSTR(SYSTEM_ALARM_OBJECT_ACE_TYPE),
+        TOSTR(ACCESS_ALLOWED_CALLBACK_ACE_TYPE),
+        TOSTR(ACCESS_DENIED_CALLBACK_ACE_TYPE),
+        TOSTR(ACCESS_ALLOWED_CALLBACK_OBJECT_ACE_TYPE),
+        TOSTR(ACCESS_DENIED_CALLBACK_OBJECT_ACE_TYPE),
+        TOSTR(SYSTEM_AUDIT_CALLBACK_ACE_TYPE),
+        TOSTR(SYSTEM_ALARM_CALLBACK_ACE_TYPE),
+        TOSTR(SYSTEM_AUDIT_CALLBACK_OBJECT_ACE_TYPE),
+        TOSTR(SYSTEM_ALARM_CALLBACK_OBJECT_ACE_TYPE),
+        TOSTR(SYSTEM_MANDATORY_LABEL_ACE_TYPE),
+    };
+#undef TOSTR
+
+    if (AceType < RTL_NUMBER_OF(AceTypes))
+        return AceTypes[AceType];
+    else
+        return "UNKNOWN TYPE";
+}
+
+/**
+ * @brief
+ * Dumps the ACE flags to the debugger output.
+ */
+static
+VOID
+SepDumpAceFlags(
+    _In_ UCHAR AceFlags)
+{
+    if (AceFlags & OBJECT_INHERIT_ACE)
+    {
+        DbgPrint("OBJECT_INHERIT_ACE\n");
+    }
+
+    if (AceFlags & CONTAINER_INHERIT_ACE)
+    {
+        DbgPrint("CONTAINER_INHERIT_ACE\n");
+    }
+
+    if (AceFlags & NO_PROPAGATE_INHERIT_ACE)
+    {
+        DbgPrint("NO_PROPAGATE_INHERIT_ACE\n");
+    }
+
+    if (AceFlags & INHERIT_ONLY_ACE)
+    {
+        DbgPrint("INHERIT_ONLY_ACE\n");
+    }
+
+    if (AceFlags & INHERITED_ACE)
+    {
+        DbgPrint("INHERITED_ACE\n");
+    }
```

Comment:
```suggestion
#define ACEPRINT(x)    \
    if (AceFlags & x) \
    { \
        DbgPrint(#x "\n"); \
    }

    ACEPRINT(OBJECT_INHERIT_ACE);
    ACEPRINT(CONTAINER_INHERIT_ACE);
    ACEPRINT(NO_PROPAGATE_INHERIT_ACE);
    ACEPRINT(INHERIT_ONLY_ACE);
    ACEPRINT(INHERITED_ACE);
#undef ACEPRINT
```
(note I didn't tested it)